### PR TITLE
virtualbox: 6.1.22 -> 6.1.26

### PR DESCRIPTION
**Pull request by parthy**
###### Motivation for this change

New release of Virtualbox with important security updates.

Note the additional change in the configure script. VirtualBox added Qt5Gui in some places that seem to break the build with a missing `<QGLWidget>`. I didn't quite understand why that doesn't work, so if there is a better way to fix the build, I'd be happy to learn.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132223` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.virtualbox</li>
    <li>linuxPackages-libre.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_4_14.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest-libre.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualboxGuestAdditions</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>linuxPackages.virtualbox (linuxPackages_5_10.virtualbox)</li>
    <li>linuxPackages_4_14.virtualbox</li>
    <li>linuxPackages_4_19.virtualbox</li>
    <li>linuxPackages_4_4.virtualbox</li>
    <li>linuxPackages_4_9.virtualbox</li>
    <li>linuxPackages_5_12.virtualbox</li>
    <li>linuxPackages_5_13.virtualbox (linuxPackages_latest.virtualbox)</li>
    <li>linuxPackages_5_4.virtualbox</li>
    <li>linuxPackages_hardened.virtualbox</li>
    <li>linuxPackages_latest_hardened.virtualbox</li>
    <li>linuxPackages_lqx.virtualbox</li>
    <li>linuxPackages_testing_bcachefs.virtualbox</li>
    <li>linuxPackages_xanmod.virtualbox</li>
    <li>linuxPackages_zen.virtualbox</li>
    <li>virtualbox</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>linuxPackages.virtualboxGuestAdditions (linuxPackages_5_10.virtualboxGuestAdditions)</li>
    <li>linuxPackages_4_14.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_19.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_4.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_9.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_12.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_13.virtualboxGuestAdditions (linuxPackages_latest.virtualboxGuestAdditions)</li>
    <li>linuxPackages_5_4.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardened.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest_hardened.virtualboxGuestAdditions</li>
    <li>linuxPackages_lqx.virtualboxGuestAdditions</li>
    <li>linuxPackages_testing_bcachefs.virtualboxGuestAdditions</li>
    <li>linuxPackages_xanmod.virtualboxGuestAdditions</li>
    <li>linuxPackages_zen.virtualboxGuestAdditions</li>
    <li>virtualboxExtpack</li>
  </ul>
</details>

**Comment by parthy** [comment]
That's odd, it builds much more for me.. And I also tested it as a custom package in the virtualbox-host module, I could run a Win10 guest just fine. Are there any logs about what went wrong?

**Comment by parthy** [comment]
I found a discussion on the vbox-dev mailing list about the compilation error: https://www.virtualbox.org/pipermail/vbox-dev/2021-July/015670.html

I've updated the PR to use that workaround and added a comment.

**Comment by siraben** [comment]
Just saw this, will rerun the review and post the failure.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132223` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.virtualbox</li>
    <li>linuxPackages-libre.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_4_14.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest-libre.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualboxGuestAdditions</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>linuxPackages.virtualbox (linuxPackages_5_10.virtualbox)</li>
    <li>linuxPackages_4_14.virtualbox</li>
    <li>linuxPackages_4_19.virtualbox</li>
    <li>linuxPackages_4_4.virtualbox</li>
    <li>linuxPackages_4_9.virtualbox</li>
    <li>linuxPackages_5_12.virtualbox</li>
    <li>linuxPackages_5_13.virtualbox (linuxPackages_latest.virtualbox)</li>
    <li>linuxPackages_5_4.virtualbox</li>
    <li>linuxPackages_hardened.virtualbox</li>
    <li>linuxPackages_latest_hardened.virtualbox</li>
    <li>linuxPackages_lqx.virtualbox</li>
    <li>linuxPackages_testing_bcachefs.virtualbox</li>
    <li>linuxPackages_xanmod.virtualbox</li>
    <li>linuxPackages_zen.virtualbox</li>
    <li>virtualbox</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>linuxPackages.virtualboxGuestAdditions (linuxPackages_5_10.virtualboxGuestAdditions)</li>
    <li>linuxPackages_4_14.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_19.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_4.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_9.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_12.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_13.virtualboxGuestAdditions (linuxPackages_latest.virtualboxGuestAdditions)</li>
    <li>linuxPackages_5_4.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardened.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest_hardened.virtualboxGuestAdditions</li>
    <li>linuxPackages_lqx.virtualboxGuestAdditions</li>
    <li>linuxPackages_testing_bcachefs.virtualboxGuestAdditions</li>
    <li>linuxPackages_xanmod.virtualboxGuestAdditions</li>
    <li>linuxPackages_zen.virtualboxGuestAdditions</li>
    <li>virtualboxExtpack</li>
  </ul>
</details>

**Comment by siraben** [comment]
```
-bash-4.2$ nix log /nix/store/ccc4aywwk4r98ira56kvqyav4vd4p5zq-virtualbox-6.1.26.drv | ix
http://ix.io/3uLQ
-bash-4.2$ nix log /nix/store/vig9shka379k7ppbzl0vahcqav7ryflx-virtualbox-6.1.26.drv | ix
http://ix.io/3uLR
```

**Comment by parthy** [comment]
Sorry about that, I should've tested the patch before.. I'm running `nixpkgs-review rev HEAD` now.

File: pkgs/applications/virtualization/virtualbox/default.nix [hunk 1]
```diff
@@ -104,6 +104,10 @@ in stdenv.mkDerivation {
     })
   ++ [
     ./qtx11extras.patch
+  ] ++ [
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virtualbox/default.nix:107]:
```suggestion
```

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132223` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.virtualbox</li>
    <li>linuxPackages-libre.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest-libre.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualboxGuestAdditions</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>linuxPackages.virtualbox</li>
    <li>linuxPackages_4_14.virtualbox</li>
    <li>linuxPackages_4_19.virtualbox</li>
    <li>linuxPackages_4_4.virtualbox</li>
    <li>linuxPackages_4_9.virtualbox</li>
    <li>linuxPackages_5_12.virtualbox</li>
    <li>linuxPackages_5_13.virtualbox</li>
    <li>linuxPackages_5_4.virtualbox</li>
    <li>linuxPackages_hardened.virtualbox</li>
    <li>linuxPackages_latest_hardened.virtualbox</li>
    <li>linuxPackages_lqx.virtualbox</li>
    <li>linuxPackages_testing_bcachefs.virtualbox</li>
    <li>linuxPackages_xanmod.virtualbox</li>
    <li>linuxPackages_zen.virtualbox</li>
    <li>virtualbox</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>linuxPackages.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_14.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_19.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_4.virtualboxGuestAdditions</li>
    <li>linuxPackages_4_9.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_12.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_13.virtualboxGuestAdditions</li>
    <li>linuxPackages_5_4.virtualboxGuestAdditions</li>
    <li>linuxPackages_hardened.virtualboxGuestAdditions</li>
    <li>linuxPackages_latest_hardened.virtualboxGuestAdditions</li>
    <li>linuxPackages_lqx.virtualboxGuestAdditions</li>
    <li>linuxPackages_testing_bcachefs.virtualboxGuestAdditions</li>
    <li>linuxPackages_xanmod.virtualboxGuestAdditions</li>
    <li>linuxPackages_zen.virtualboxGuestAdditions</li>
    <li>virtualboxExtpack</li>
  </ul>
</details>

**Comment by parthy** [comment]
@siraben The logs for those failed builds would be interesting, I could build it without problems.

**Comment by siraben** [comment]
http://ix.io/3uMg , the other failures are identical.

**Comment by parthy** [comment]
> http://ix.io/3uMg , the other failures are identical.

Could this line be the issue in your case?

```
qemu-i386: Unable to reserve 0xfffff000 bytes of virtual address space at 0x1000 (Success) for use as guest address space (check yourvirtual memory ulimit setting, min_mmap_addr or reserve less using -R option)
```

**Comment by siraben** [comment]
Ah, I've seen that error on my builder before, ok, disregard my build logs please.

**Comment by ilya-fedin** [comment]
Am I the only who experience `NS_ERROR_FAILURE (0x80004005)` when starting virtual machines after this update?
I have this in dmesg after each try to launch a VM:
```
[41222.555383] vboxdrv: 0000000000000000 VMMR0.r0
[41222.629042] vboxdrv: 0000000000000000 VBoxDDR0.r0
[41222.696571] VMSVGA FIFO[55108]: segfault at f8 ip 00007f7d54346ab9 sp 00007f7cba11cbe0 error 4 in libX11.so.6.4.0[7f7d54342000+8c000]
[41222.696598] Code: 74 05 48 89 ef ff d0 48 83 c4 08 b8 01 00 00 00 5b 5d c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 40 00 55 48 89 fd 53 48 83 ec 08 <48> 8b 87 f8 00 00 00 a8 02 0f 85 a2 00 00 00 48 8b 87 f8 00 00 00
```

**Comment by raboof** [comment]
I'm also seeing problems. On the commit of this upgrade, the `VirtualBox` UI shows:

```
Failed to construct device 'ichac97' instance #0 (VERR_INVALID_PARAMETER)
```

... for me. Since ich ac97 is the audio card, I tried disabling audio for this image and that seems to resolve the failure.

**Comment by parthy** [comment]
That's a separate issue and actually started before: https://github.com/NixOS/nixpkgs/issues/123851#issuecomment-890929628

**Comment by ilya-fedin** [comment]
> I tried disabling audio for this image and that seems to resolve the failure.

Thanks, it helped me as well

**Pull request merged**